### ISSUE TITLE
训练出来的模型与论文结果偏差过大，mae达到了15点多，以下是我数据处理的部分

### ISSUE THREAD

**Issue by zhuwei321** [ICDM-UESTC/MMRA#6]
`import pandas as pd
import pickle
import numpy as np
from tqdm import tqdm
from textual_engineering import load_angle_bert_model, angle_bert_textual_feature_extraction
from visual_engineering import load_vit_model, vit_visual_feature_extraction
import os

# 加载 data.pkl
data_path = "/home/zhuwei/MMRA-main/data/MicroLens-100k/data.pkl"
with open(data_path, "rb") as f:
    data = pickle.load(f)

# 加载模型
angle_model = load_angle_bert_model()
vit_processor, vit_model = load_vit_model()

# 提取文本特征
textual_features = []
for text in tqdm(data["text"]):  # 替换为实际的文本列名
    textual_features.append(angle_bert_textual_feature_extraction(angle_model, text))

# 添加文本特征到 DataFrame
data["textual_feature_embedding"] = textual_features

# 提取视觉特征
visual_features_cls = []
visual_features_mean = []
image_base_path = "/mnt/myusb2/MicroLens-100k/videos-frame/"  # 图片所在的目录路径

# 零向量定义
zero_vector = [0.0] * 768  # 假设视觉特征维度为 768

for video_id in tqdm(data["item_id"]):  # 遍历视频 ID 列
    video_features = []
    for i in range(10):  # 每个视频有 10 张图片
        image_path = f"{image_base_path}{video_id}_{i}.jpg"  # 构建图片路径
        if os.path.exists(image_path):  # 检查图片文件是否存在
            try:
                feature = vit_visual_feature_extraction(vit_processor, vit_model, image_path)
                if feature is not None:  # 检查是否成功提取特征
                    video_features.append(feature)
                else:
                    video_features.append(zero_vector)  # 无效特征填充零向量
            except Exception as e:
                video_features.append(zero_vector)  # 如果出现错误，填充零向量
        else:
            video_features.append(zero_vector)  # 如果图片不存在，填充零向量

    # 确保每个视频都有 10 个特征
    while len(video_features) < 10:
        video_features.append(zero_vector)

    # 将 10 张图片的视觉特征保存到 visual_feature_embedding_cls 列
    visual_features_cls.append(video_features)
    # 计算每个视频的平均视觉特征，并保存到 visual_feature_embedding_mean 列
    avg_visual_features = np.mean(video_features, axis=0).tolist()
    visual_features_mean.append(avg_visual_features)

# 添加视觉特征到 DataFrame
data["visual_feature_embedding_cls"] = visual_features_cls
data["visual_feature_embedding_mean"] = visual_features_mean

# 保存回 data.pkl
with open(data_path, "wb") as f:
    pickle.dump(data, f)

print("特征提取完成，数据已保存！")`

**Comment by Jian-Lang** [comment]
Hi, Thank you for your interest in our work.

From your email, I noticed that you reported an MAE of 15 and an nMSE of 0.4. In comparison, our published results showed an MAE of 1.17 and an nMSE of 0.75. While your nMSE appears better than ours, your MAE is significantly higher. This discrepancy suggests there might be an issue with your training or testing process.

Since your shared code for data preprocessing looks correct and the key processing steps rely on the code I provided, this part should not have any major issues. I’d recommend double-checking other aspects of your pipeline, particularly:

+ Training/Testing Split: Ensure the data splits are consistent with those described in the paper.
Hyperparameter Settings: Verify that your hyperparameters (e.g., learning rate, epochs, batch size) match those we used in the experiments.

+ Hyperparameter Settings: Verify that your hyperparameters (e.g., learning rate, epochs, batch size) match those we used in the experiments.

Hope this can help you. Feel free to contact again.

**Comment by zhuwei321** [comment]
感谢您的回复，经过我的检查，超参数都是按您的默认设置，应该是一致的。数据集分割也是调用您提供的代码实现的。
以下是我数据处理的第二部分的代码，您看看是否有误：
`# 加载数据
data_path = "/home/zhuwei/MMRA-main/data/MicroLens-100k/data.pkl"
with open(data_path, "rb") as f:
    data = pickle.load(f)

# 加载 BLIP 模型
image_processor, image_model = loading_model()

# 加载 AngleBERT 模型
angle_model = load_angle_model()

# 视频帧字幕生成和检索向量计算
retrieval_features = []

# 视频帧的图像标题提取和拼接
image_base_path = "/mnt/myusb2/MicroLens-100k/videos-frame/"  # 图片所在的目录路径

for item_id in tqdm(data["item_id"]):  # 遍历每个视频 ID

    frame_texts = []

    for j in range(0, 10):

        image_path = os.path.join(image_base_path, f"{item_id}_{j}.jpg")
        if os.path.exists(image_path):
          try:
            text = convert_image_to_text(image_processor, image_model, image_path)
          except Exception as e:
            continue
        else:
            continue
        frame_texts .append(text)

    # 合成视频标题（原始标题 + 帧字幕）
    video_title = data.loc[data['item_id'] == item_id, 'text'].values[0]  # 获取原始视频标题
    combined_text = " ".join(frame_texts) + " " + video_title  # 拼接视频帧标题和原始标题

    # 获取检索向量
    retrieval_vector = convert_text_to_embedding(angle_model, combined_text)

    # 添加检索向量到列表
    retrieval_features.append(retrieval_vector[0].tolist())  # 取出numpy数组并转换为列表

# 将检索向量添加到数据中
data["retrieval_feature"] = retrieval_features

# 保存更新后的数据
with open(data_path, "wb") as f:
    pickle.dump(data, f)

print("数据更新完成，检索向量已添加到 data.pkl！")
`

**Comment by Jian-Lang** [comment]
It seems that you leveraged the on-screen text. However, I do not mention to use this. So I guess this may lead unpredictable result. Hope this can help you.

**Comment by zhuwei321** [comment]
[微信图片_20241214092813.pdf](https://github.com/user-attachments/files/18133827/_20241214092813.pdf)
In your paper, you did mention extracting subtitles from frames. I wonder if there is a problem with the processing method.

**Comment by Jian-Lang** [comment]
Hi, your understanding is wrong. In the picture you attached, the textual description is the title of the video. So no subtitles are used in our work and I also do not provide the relevant code indeed.

**Comment by zhuwei321** [comment]
感谢你的回答

**Comment by Jian-Lang** [comment]
@skyandkibo Sorry for late response. Yes, you need to first concat the video caption with the title and feed the composed one for text_semantic_embedding to perform retrieval.
